Mathlib/Data/ENNReal/Inv.lean

```diff
@@ -0,0 +1,679 @@
+/-
+Copyright (c) 2017 Johannes Hölzl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johannes Hölzl, Yury Kudryashov
+-/
+import Mathlib.Data.ENNReal.Operations
+
+/-!
+# Results about division in extended non-negative reals
+
+This file places `Inv` and `Div` instances on `ℝ≥0∞` making it into a `DivInvOneMonoid` and
+establishes basic properties related to these operations. As a consequence of being a
+`DivInvOneMonoid`, `ℝ≥0∞` inherits a power operation with integer exponent.
+
+## Main definitions
+
+  - `a⁻¹` is defined as `Inf {b | 1 ≤ a * b}`. This way we have `(↑p)⁻¹ = ↑(p⁻¹)` for
+    `p : ℝ≥0`, `p ≠ 0`, `0⁻¹ = ∞`, and `∞⁻¹ = 0`.
+
+  - `a / b` is defined as `a * b⁻¹`.
+
+A few order isomorphisms are worthy of mention:
+
+  - `OrderIso.invENNReal : ℝ≥0∞ ≃o ℝ≥0∞ᵒᵈ`: The map `x ↦ x⁻¹` as an order isomorphism to the dual.
+
+  - `orderIsoIicOneBirational : ℝ≥0∞ ≃o Iic (1 : ℝ≥0∞)`: The birational order isomorphism between
+    `ℝ≥0∞` and the unit interval `Set.Iic (1 : ℝ≥0∞)` given by `x ↦ (x⁻¹ + 1)⁻¹` with inverse
+    `x ↦ (x⁻¹ - 1)⁻¹`
+
+  - `orderIsoIicCoe (a : ℝ≥0) : Iic (a : ℝ≥0∞) ≃o Iic a`: Order isomorphism between an initial
+    interval in `ℝ≥0∞` and an initial interval in `ℝ≥0` given by the identity map.
+
+  - `orderIsoUnitIntervalBirational : ℝ≥0∞ ≃o Icc (0 : ℝ) 1`: An order isomorphism between
+    the extended nonnegative real numbers and the unit interval. This is `orderIsoIicOneBirational`
+    composed with the identity order isomorphism between `Iic (1 : ℝ≥0∞)` and `Icc (0 : ℝ) 1`.
+-/
+
+open Set BigOperators NNReal ENNReal
+
+namespace ENNReal
+
+noncomputable section Inv
+
+variable {a b c d : ℝ≥0∞} {r p q : ℝ≥0}
+
+-- porting note: rfc: redefine using pattern matching?
+instance : Inv ℝ≥0∞ := ⟨fun a => sInf { b | 1 ≤ a * b }⟩
+
+instance : DivInvMonoid ℝ≥0∞ where
```

Comment:
I think this should be moved to the beginning(-ish) of the `Basic` file. It would be nice if all the data-bearing classes are declared there.

Reply:
Do you mean just this instance? (I'm not sure what "all the data-bearing classes" includes.)